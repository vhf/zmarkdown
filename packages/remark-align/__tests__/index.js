import dedent from 'dedent'
import unified from 'unified'
import reParse from 'remark-parse'
import stringify from 'rehype-stringify'
import remark2rehype from 'remark-rehype'

import remarkAlign from '../src/'

const render = (text, config) => unified()
  .use(reParse)
  .use(remarkAlign, config)
  .use(remark2rehype)
  .use(stringify)
  .processSync(text)

const alignFixture = dedent`
  Test align
  ==========

  A simple paragraph

  ->A centered paragraph<-

  a simple paragraph

  ->A right aligned paragraph->

  an other simple paragraph

  A simple paragraph

  ->A centered paragraph<-

  a simple paragraph

  ->A right aligned paragraph->

  an other simple paragraph

  ->A centered paragraph.

  Containing two paragraph<-

  an other simple paragraph

  ->A right aligned paragraph.

  Containing two paragraph<-

  an other simple paragraph

  ->A centered paragraph.<-
  ->An other centered paragraph.<-

  a simple paragraph

  ->A started block without end.
`

test('align', () => {
  const { contents } = render(alignFixture)
  expect(contents).toMatchSnapshot()
})

test('align-custom-config', () => {
  const { contents } = render(alignFixture, {
    right: 'custom-right',
    center: 'custom-center'
  })
  expect(contents).toMatchSnapshot()
})

test('block-wrap', () => {
  const { contents } = render(dedent`
    # wraps blocks e.g. title:

    -> foo

    # title

    foo ->
  `)
  expect(contents).toMatchSnapshot()
})

test('center-no-start', () => {
  const { contents } = render(dedent`
    # title

    foo <-

    # title
  `)
  expect(contents).toMatchSnapshot()
})

test('right-no-end', () => {
  const { contents } = render(dedent`
    # title

    -> foo

    # title
  `)
  expect(contents).toMatchSnapshot()
})

test('right-no-start', () => {
  const { contents } = render(dedent`
    # title

    foo ->

    # title
  `)
  expect(contents).toMatchSnapshot()
})
