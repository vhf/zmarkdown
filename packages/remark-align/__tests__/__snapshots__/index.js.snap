// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`align 1`] = `
"<h1>Test align</h1>
<p>A simple paragraph</p>
<div class=\\"align-center\\"><p>A centered paragraph</p></div>
<p>a simple paragraph</p>
<div class=\\"align-right\\"><p>A right aligned paragraph</p></div>
<p>an other simple paragraph</p>
<p>A simple paragraph</p>
<div class=\\"align-center\\"><p>A centered paragraph</p></div>
<p>a simple paragraph</p>
<div class=\\"align-right\\"><p>A right aligned paragraph</p></div>
<p>an other simple paragraph</p>
<div class=\\"align-center\\"><p>A centered paragraph.</p><p>Containing two paragraph</p></div>
<p>an other simple paragraph</p>
<div class=\\"align-center\\"><p>A right aligned paragraph.</p><p>Containing two paragraph</p></div>
<p>an other simple paragraph</p>
<div class=\\"align-center\\"><p>A centered paragraph.</p><p>An other centered paragraph.</p></div>
<p>a simple paragraph</p>
<p>->A started block without end.</p>"
`;

exports[`align-custom-config 1`] = `
"<h1>Test align</h1>
<p>A simple paragraph</p>
<div class=\\"custom-center\\"><p>A centered paragraph</p></div>
<p>a simple paragraph</p>
<div class=\\"custom-right\\"><p>A right aligned paragraph</p></div>
<p>an other simple paragraph</p>
<p>A simple paragraph</p>
<div class=\\"custom-center\\"><p>A centered paragraph</p></div>
<p>a simple paragraph</p>
<div class=\\"custom-right\\"><p>A right aligned paragraph</p></div>
<p>an other simple paragraph</p>
<div class=\\"custom-center\\"><p>A centered paragraph.</p><p>Containing two paragraph</p></div>
<p>an other simple paragraph</p>
<div class=\\"custom-center\\"><p>A right aligned paragraph.</p><p>Containing two paragraph</p></div>
<p>an other simple paragraph</p>
<div class=\\"custom-center\\"><p>A centered paragraph.</p><p>An other centered paragraph.</p></div>
<p>a simple paragraph</p>
<p>->A started block without end.</p>"
`;

exports[`block-wrap 1`] = `
"<h1>wraps blocks e.g. title:</h1>
<div class=\\"align-right\\"><p> foo</p><h1>title</h1><p>foo </p></div>"
`;

exports[`center-no-start 1`] = `
"<h1>title</h1>
<p>foo &#x3C;-</p>
<h1>title</h1>"
`;

exports[`right-no-end 1`] = `
"<h1>title</h1>
<p>-> foo</p>
<h1>title</h1>"
`;

exports[`right-no-start 1`] = `
"<h1>title</h1>
<p>foo -></p>
<h1>title</h1>"
`;
